--- discover.php.old	2009-03-19 15:16:04.000000000 -0700
+++ discover.php	2009-03-19 14:01:44.000000000 -0700
@@ -296,7 +296,7 @@
 		<input type=hidden name=action value="save">
 		<input type=hidden name=hostname value=\'' . $row['ip'] . "'>
 		<input type=hidden name=id value=0>
-		<input type=hidden name=description value='" . $row['hostname'] . "'>
+		<input type=hidden name=description value=''>
 		<input type=hidden name=snmp_community value='" . $row['community'] . "'>
 		<input type=hidden name=snmp_version value='" . $row['snmp_version'] . "'>
 		<input type=hidden name=snmp_username value='" . $row['snmp_username'] . "'>
