--- findhosts.php	2009-03-19 12:06:33.000000000 -0700
+++ fh.php	2009-03-19 12:39:25.000000000 -0700
@@ -355,7 +355,7 @@
 								. sql_sanitize($device['snmp_sysDescr']) . "', '"
 								. sql_sanitize($device['snmp_sysUptime']) . "', '"
 								. sql_sanitize($device['os']) . "', "
-								. "1, 1,".time() . ')' );
+								. "'1','1','" .time() . "')");
 						}
 					}
 				} else {
@@ -390,9 +390,9 @@
 								. sql_sanitize($device['snmp_sysName']) . "', '"
 								. sql_sanitize($device['snmp_sysLocation']) . "', '"
 								. sql_sanitize($device['snmp_sysContact']) . "', '"
-								. sql_sanitize($device['snmp_sysDescr']) . "', "
-								. sql_sanitize($device['snmp_sysUptime']) . ", "
-								. "'', 0, 1,".time() . ')' );
+								. sql_sanitize($device['snmp_sysDescr']) . "', '"
+								. sql_sanitize($device['snmp_sysUptime']) . "'"
+								. ",'0','1','" .time() . "')");
 							discover_debug(" - Host is alive but no SNMP!");
 						} else {
 						}
