diff -urN cacti-0.8.7e.orig/include/global_arrays.php cacti-0.8.7e/include/global_arrays.php
--- cacti-0.8.7e.orig/include/global_arrays.php	2009-06-28 09:07:11.000000000 -0700
+++ cacti-0.8.7e/include/global_arrays.php	2010-03-18 10:46:18.248313117 -0700
@@ -262,7 +262,8 @@
 $rrdtool_versions = array(
 	"rrd-1.0.x" => "RRDTool 1.0.x",
 	"rrd-1.2.x" => "RRDTool 1.2.x",
-	"rrd-1.3.x" => "RRDTool 1.3.x");
+	"rrd-1.3.x" => "RRDTool 1.3.x",
+	"rrd-1.4.x" => "RRDTool 1.4.x");
 
 $cdef_item_types = array(
 	1 => "Function",
diff -urN cacti-0.8.7e.orig/install/index.php cacti-0.8.7e/install/index.php
--- cacti-0.8.7e.orig/install/index.php	2009-06-28 09:07:11.000000000 -0700
+++ cacti-0.8.7e/install/index.php	2010-03-18 10:47:29.082310792 -0700
@@ -282,7 +282,9 @@
 	exec($input["path_rrdtool"]["default"], $out_array);
 
 	if (sizeof($out_array) > 0) {
-		if (ereg("^RRDtool 1\.3", $out_array[0])) {
+		if (ereg("^RRDtool 1\.4", $out_array[0])) {
+			$input["rrdtool_version"]["default"] = "rrd-1.4.x";
+		}else if (ereg("^RRDtool 1\.3\.", $out_array[0])) {
 			$input["rrdtool_version"]["default"] = "rrd-1.3.x";
 		}else if (ereg("^RRDtool 1\.2\.", $out_array[0])) {
 			$input["rrdtool_version"]["default"] = "rrd-1.2.x";
diff -urN cacti-0.8.7e.orig/utilities.php cacti-0.8.7e/utilities.php
--- cacti-0.8.7e.orig/utilities.php	2009-06-28 09:07:11.000000000 -0700
+++ cacti-0.8.7e/utilities.php	2010-03-18 10:50:08.273311540 -0700
@@ -249,7 +249,9 @@
 		exec(read_config_option("path_rrdtool"), $out_array);
 
 		if (sizeof($out_array) > 0) {
-			if (ereg("^RRDtool 1\.3", $out_array[0])) {
+			if (ereg("^RRDtool 1\.4", $out_array[0])) {
+				$rrdtool_version = "rrd-1.4.x";
+			}else if (ereg("^RRDtool 1\.3\.", $out_array[0])) {
 				$rrdtool_version = "rrd-1.3.x";
 			}else if (ereg("^RRDtool 1\.2\.", $out_array[0])) {
 				$rrdtool_version = "rrd-1.2.x";
