--- lib/platform/CXWindowsScreen.cpp.save	Sun Apr 20 14:32:02 2008
+++ lib/platform/CXWindowsScreen.cpp	Sun Apr 20 14:34:33 2008
@@ -34,8 +34,11 @@
 #	include <X11/Xutil.h>
 #	define XK_MISCELLANY
 #	define XK_XKB_KEYS
 #	include <X11/keysymdef.h>
+	extern "C" {
+#	include <X11/extensions/dpms.h>
+	}
 #	if HAVE_X11_EXTENSIONS_XTEST_H
 #		include <X11/extensions/XTest.h>
 #	else
 #		error The XTest extension is required to build synergy
@@ -228,8 +231,18 @@
 		CXWindowsUtil::CErrorLock lock(m_display);
 		XSetInputFocus(m_display, m_lastFocus, m_lastFocusRevert, CurrentTime);
 	}
 
+	// Force the DPMS to turn screen back on since we don't
+	// actually cause physical hardware input to trigger it
+	int dummy;
+	if (DPMSQueryExtension(m_display, &dummy, &dummy))
+	{
+		DPMSEnable(m_display);
+		DPMSForceLevel(m_display, DPMSModeOn);
+		DPMSEnable(m_display);
+	}
+
 	// unmap the hider/grab window.  this also ungrabs the mouse and
 	// keyboard if they're grabbed.
 	XUnmapWindow(m_display, m_window);
 
